packages/sdk/src/types.ts

```diff
@@ -127,23 +151,29 @@ export interface RangeEditionSchedule extends MintScheduleBase {
   cutoffTime: number
   maxMintable: (unixTimestamp?: number) => number
 }
+export interface RangeEditionV1Schedule extends RangeEditionSchedule {
+  interfaceId: typeof interfaceIds.IRangeEditionMinter
+}
+export interface RangeEditionV2Schedule extends RangeEditionSchedule {
+  interfaceId: typeof interfaceIds.IRangeEditionMinterV2
+}
 
 export interface MerkleDropSchedule extends MintScheduleBase {
   mintType: 'MerkleDrop'
   maxMintable: number
   merkleRoot: string
 }
-
-export type MintSchedule = RangeEditionSchedule | MerkleDropSchedule
-
-export function isRangeEditionSchedule(schedule: MintSchedule): schedule is RangeEditionSchedule {
-  return schedule.mintType === 'RangeEdition'
+export interface MerkleDropV1Schedule extends MerkleDropSchedule {
+  interfaceId: typeof interfaceIds.IMerkleDropMinter
 }
-
-export function isMerkleDropSchedule(schedule: MintSchedule): schedule is MerkleDropSchedule {
```

Comment:
any reason we are removing these helpers?

Reply:
trivially identified with schedule.mintType === 'RangeEdition' and annoying to add one for each minterType / version type

Reply:
1. removing these helpers is a potential breaking change.
2. I would like this function to tell me if given schedule is range or merkle or not. This way regardless of version we can return

Reply:
Re1, there was only on place that used this that was trivially replaced by schedule.mintType === 'RangeEdition"

Re 2, you know that it is range or merkle because the schedule has a typed paramater called mintType, this is literally why that typed string exists in the schedule type


Open to readding if needed, but how should I resolve, 
![image](https://user-images.githubusercontent.com/3156233/235395780-908a1c57-99ea-4220-83c4-27a9901f6805.png)

Reply:
we should keep the type guard, it's part of how typescript works to be able to leverage helpers

Reply:
what am I missing here? 

this was the only usage I saw:

![image](https://user-images.githubusercontent.com/3156233/235414459-c6b37d8f-6dd1-4db3-815a-304012cbc942.png)

Reply:
I feel strongly on this because it's part of the sdk giving type guards to make the usage simpler, core typescript helpers like  `.filter` rely on type guards just to give an example

Reply:
it's not just where it's used, it's where it `can` be used